design-toolkit/components/scripts/generate-tokens.mjs

```diff
@@ -10,165 +10,42 @@
-      __dirname,
-      '..',
-      'src',
-      'tokens',
-      'semantic.json',
-    );
-    const semanticContent = readFileSync(semanticPath, 'utf-8');
-    const semantic = JSON.parse(semanticContent);
-    // Generate files in the src/tokens directory
-    const outputDir = path.join(__dirname, '..', 'src', 'tokens');
-
-    await fs.promises.mkdir(outputDir, { recursive: true });
-
-    // Generate CSS variables
-    const cssContent = generateCSS(tokens);
-
-    await fs.promises.writeFile(
-      path.join(outputDir, 'tokens.css'),
-      cssContent,
-      'utf-8',
-    );
-
-    // Generate TypeScript constants and types
-    const tsContent = generateTypeScript(tokens, semantic);
-
-    await fs.promises.writeFile(
-      path.join(outputDir, 'index.ts'),
-      tsContent,
-      'utf-8',
-    );
-
-    // Generate theme blocks CSS
-    const themesContent = generateThemesCSS(tokens, semantic);
-
-    await fs.promises.writeFile(
-      path.join(outputDir, 'themes.css'),
-      themesContent,
-      'utf-8',
-    );
+const OUTPUT_DIR = path.join(__dirname, '..', 'src', 'tokens');
+const INPUT_DIR = path.join(__dirname, '..', 'src', 'tokens');
 
-    console.log('✅ Design tokens generated successfully!');
-  } catch (error) {
-    console.error('❌ Error generating design tokens:', error);
-    process.exit(1);
-  }
-}
-
-function flattenTokens(obj, prefix = '') {
-  const result = {};
-  for (const [key, value] of Object.entries(obj)) {
-    const newKey = prefix ? `${prefix}-${key}` : key;
-    if (typeof value === 'object' && value !== null) {
-      Object.assign(result, flattenTokens(value, newKey));
-    } else {
-      result[newKey] = value;
-    }
-  }
-  return result;
-}
-
-function toCamelCase(str) {
-  // Remove all non-alphanumeric characters and convert to camelCase
-  return str
-    .replace(/[-_]+(.)?/g, (_, chr) => (chr ? chr.toUpperCase() : ''))
-    .replace(/^(\d)/, '_$1'); // prefix leading digit with underscore
-}
-
-function generateCSS(tokens) {
-  const flattened = flattenTokens(tokens);
-  const cssLines = [];
-
-  // Start :root block
-  cssLines.push(':root {');
-
-  // Generate CSS variables
-  for (const [key, value] of Object.entries(flattened)) {
-    cssLines.push(`  --${key}: ${value};`);
-  }
-
-  // Close :root block
-  cssLines.push('}');
-
-  return cssLines.join('\n');
-}
-
-function extractVarReference(value) {
-  // Check if the value is a string
-  if (typeof value !== 'string') {
-    return null;
-  }
-
-  // Check if the value is a var(--...) reference
-  const varMatch = value.match(/^var\(--([a-zA-Z0-9-_]+)\)$/);
+const APPLIED_PROPERTY_MAP = {
+  bg: ['background-color'],
+  fg: ['color', '--icon-color'],
+  outline: ['outline-color'],
+  shadow: ['box-shadow'],
+};
+const skipFallback = ['icon-size', 'shadow-elevation', 'font'];
 
-  // Return the matched variable name or null if not found
-  return varMatch ? varMatch[1] : null;
+//#region I/O utils
```

Comment:
i tried to organize things in this file a little bit
 
region comments make bigger files like this a little easier to navigate.
<img width="1464" height="1207" alt="Screenshot 2025-08-22 at 3 31 40 PM" src="https://github.com/user-attachments/assets/3da152de-9069-4c44-a38c-43b6b5660a4e" />

i think ideally, we'd move some of this stuff out to utils, and potentially even expose some things to apps (like the conversion utils).

also would be nice to write this in typescript eventually.